news/10222.bugfix.rst

```diff
@@ -0,0 +1 @@
+Fix ``pip install --pre`` for packages with pre-release dependencies defined both in ``pyproject.toml`` and ``setup.py``.
```

Comment:
This is ambigious because you can define runtime dependencies in `pyproject.toml` as well. It’s better to say “build dependencies” instead, like you did in the PR description.